go-controller/pkg/ovn/egressgw_test.go

```diff
@@ -2087,7 +2087,185 @@ var _ = ginkgo.Describe("OVN Egress Gateway Operations", func() {
+				return nil
+			}
+
+			err := app.Run([]string{app.Name})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		})
+		ginkgo.It("validates that BFD Rate limiter decreases proportionally to number of BFD connections", func() {
+			app.Action = func(ctx *cli.Context) error {
+				staticRoutes := []string{"10.0.0.1"}
+				namespaceT := *newNamespace(namespaceName)
+				namespaceX := *newNamespace("namespace2")
+				gr := &nbdb.LogicalRouter{
+					UUID: "GR_node1-UUID",
+					Name: "GR_node1",
+				}
+				bfdBand := &nbdb.MeterBand{
+					UUID:        "bfdUUID",
+					Action:      nbdb.MeterBandActionDrop,
+					Rate:        types.BFDRateLimit,
+					ExternalIDs: map[string]string{"bfd-" + types.OvnRateLimitingMeter: "true"},
+				}
+				t := newTPod("node1", "10.128.1.0/24", "10.128.1.2", "10.128.1.1", "myPod", "10.128.1.3", "0a:58:0a:80:01:03", namespaceT.Name)
+				gwPod := *newPod(namespaceX.Name, "gwPod", "node2", staticRoutes[0])
+				gwPod.Annotations = map[string]string{"k8s.ovn.org/routing-namespaces": namespaceT.Name}
+				gwPod.Annotations["k8s.ovn.org/bfd-enabled"] = ""
+				gwPod.Spec.HostNetwork = true
+				pods := &v1.PodList{
+					Items: []v1.Pod{
+						*newPod(t.namespace, t.podName, t.nodeName, t.podIP),
+						gwPod,
+					},
+				}
+				fakeOvn.startWithDBSetup(
+					libovsdbtest.TestSetup{
+						NBData: []libovsdbtest.TestData{
+							&nbdb.LogicalSwitch{
+								UUID: "node1",
+								Name: "node1",
+							},
+							&nbdb.LogicalSwitch{
+								UUID: "node2",
+								Name: "node2",
+							},
+							bfdBand,
+							&nbdb.Meter{
+								UUID:  "meterUUID",
+								Bands: []string{"bfdUUID"},
+							},
+							gr,
+						},
+					},
+					&v1.NamespaceList{
+						Items: []v1.Namespace{
+							namespaceT,
+							namespaceX,
+						},
+					},
+					&v1.NodeList{
+						Items: []v1.Node{
+							*newNode("node1", "192.168.126.202/24"),
+							*newNode("node2", "10.128.2.0/24"),
+						},
+					},
+					pods,
+				)
+				t.populateLogicalSwitchCache(fakeOvn)
+				fakeOvn.controller.lsManager.AddOrUpdateSwitch("node2", []*net.IPNet{ovntest.MustParseIPNet("10.128.2.0/24")})
+				injectNode(fakeOvn)
+				err := fakeOvn.controller.WatchNamespaces()
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = fakeOvn.controller.WatchPods()
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				gr.StaticRoutes = []string{"static-route-1-UUID"}
+				finalDB := []libovsdbtest.TestData{
+					&nbdb.LogicalSwitchPort{
+						UUID:      "lsp1",
+						Addresses: []string{"0a:58:0a:80:01:03 10.128.1.3"},
+						ExternalIDs: map[string]string{
+							"pod":       "true",
+							"namespace": namespaceName,
+						},
+						Name: "namespace1_myPod",
+						Options: map[string]string{
+							"iface-id-ver":      "myPod",
+							"requested-chassis": "node1",
+						},
+						PortSecurity: []string{"0a:58:0a:80:01:03 10.128.1.3"},
+					},
+					&nbdb.LogicalSwitch{
+						UUID:  "node1",
```

Comment:
nit: here and in a few places. you don't use the standard conventions for creating UUIDs. I won't block if this needs to get in

Reply:
hmm I see this being a convention in all other places on this file; but I am happy to change this to the standard convention. could you please point me to the util you have in mind that i should be using?

Reply:
it is not that big of deal but it seems like most other test files use something like "node1-UUID", so "object-UUID" . I think that is to differentiate the name from the UUID.  I wish there was a series of utils to build these

Reply:
ah I see, yup will change these.